### ISSUE TITLE
Setting text of textfield outside of masked area flashes visible briefly

### ISSUE THREAD

**Issue by mayorbyrne** [bp74/StageXL#299]
I wrote some code simulating button presses, which add sprites to a masked-off wrapper. The sprite contains a TextField. After the sprite is added, I loop through the list of sprites and update the text of the TextFields inside them. In doing so, the sprites outside the viewable area appear to flash "visible" for a brief second before becoming masked off again. I haven't quite figured out exactly which part(s) of the sprite display viewable...in this particular case, I have a Shape and a TextField as the children, and regardless of child order, only the shape is flashing visible outside of the viewable area. If the TextField is the sole child, it flashes visible. This is a regression from StageXL 1.2.0.

Sorry if this is confusing. [I recorded my screen to better show what's happening, using v1.3.0](https://github.com/bp74/StageXL/files/1703413/stagexl.textfield.bug.zip)

Also, here is the code.

```js
import 'dart:html' as html;
import 'package:stagexl/stagexl.dart';
import 'dart:async';

Sprite _wrapper;
int _posY;
List<Sprite> _items;

void main()
{
  StageOptions options = new StageOptions()
    ..stageAlign = StageAlign.TOP_LEFT
    ..stageScaleMode = StageScaleMode.NO_SCALE
    ..renderEngine = RenderEngine.WebGL;

  Stage stage = new Stage(
      html.querySelector("#stage"), width: 990, height: 620, options: options);

  RenderLoop renderLoop = new RenderLoop()
    ..addStage(stage);

  _items = [];

  Sprite bg = new Sprite()
    ..graphics.rect(0, 0, 500, 500)
    ..graphics.strokeColor(Color.LightGray, 2)
    ..x = 50
    ..y = 50
    ..addTo(stage);

  _wrapper = new Sprite()
    ..addTo(bg);

  Mask wrapperMask = new Mask.rectangle(0, 0, 500, 500);
  wrapperMask.relativeToParent = true;
  _wrapper.mask = wrapperMask;

  _posY = 0;

  // "Click" button 30 times
  int count = 0;
  new Timer.periodic(new Duration(milliseconds: 300), (Timer timer){
    count++;
    _simulateButtonClick();
    if (count == 30) timer.cancel();
  });
}

_simulateButtonClick()
{
  Shape shape = new Shape()
    ..graphics.rect(0, 0, 300, 22)
    ..graphics.strokeColor(Color.Black);
  TextField x = new TextField("Sample", new TextFormat("Arial", 12, Color.Black))
    ..width = 300
    ..height = 22;

  Sprite sprite = new Sprite()
    ..addChild(shape)
    ..addChild(x)
    ..x = 10
    ..y = _posY + 2
    ..addTo(_wrapper);

  _items.add(sprite);

  _posY += 52;

  _resetText();
}

_resetText()
{
  int count = 0;
  for (Sprite item in _items)
  {
    TextField childText = item.children[1];
    childText.text = "$count.";
    count++;
  }
}
```

**Comment by bp74** [comment]
wow that is strange! i will take a look at it.

**Comment by bp74** [comment]
okay i know what is going on but the fix needs a little bit more time. It's again the scissors optimization of rectangular masks, this time in combination with a texture upload after changing the texture of a TextField.

**Comment by bp74** [comment]
Fixed it! Could you please do another test with the master branch from GitHub? Thanks!

**Comment by bp74** [comment]
Great! I've released StageXL 1.3.1+1. Thanks for reporting those bugs!
